server/src/main/java/tw/waterball/survive/io/server/nio/NioServer.java

```diff
@@ -0,0 +1,98 @@
+package tw.waterball.survive.io.server.nio;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.*;
+
+import java.util.Iterator;
+
+/**
+ * @author swshawnwu@gmail.com(ShawnWu)
+ */
+public class NioServer {
+
+    private ServerSocketChannel socketChannel = ServerSocketChannel.open();
+    private Selector selector = Selector.open();
+    private SelectionKey selection;
+    private ByteBuffer buffer;
+    private boolean serverStatus;
+
+    private static NioServer nioServer;
+
+    public static NioServer create(int port) throws IOException {
+        if (nioServer == null)
+            nioServer = new NioServer(port);
+        return nioServer;
+    }
+
+    private NioServer(int port) throws IOException {
+        socketChannel.configureBlocking(false);
+        socketChannel.socket().bind(new InetSocketAddress(port));
+        socketChannel.register(selector, SelectionKey.OP_ACCEPT);
+    }
+
+    public void start() throws IOException {
+        serverStatus = true;
+        while (serverStatus) {
```

Comment:
Rename it to running or whatever better to explain its meaning.
serverStatus is just like data, info which are not clear.

Reply:
Nice suggestion, I'll adopt it.